[DI-2709] Adds CI worflow for the generated project (#89)
django-stars/backend-skeleton: {{ cookiecutter.project_slug }}/docs/pre_commit_hook.md

To run additional checks before making commit we use [Pre-commit](https://pre-commit.com/) hooks.

> Since the `api` is running inside docker container we don't have a pre-commit as a dependency there, so you need to 
> install `pre-commit` package [manually](https://pre-commit.com/#install).

Reviewer: I think this make more sense in this way. Not sure about "locally" but I want to emphasize that pre-commit needs to be installed system wise 🤔 
```suggestion
Since the `api` is running inside docker container and we don't have a pre-commit as a dependency there, you need to 
install `pre-commit` package [locally](https://pre-commit.com/#install).
```

> [!NOTE]  

> > The `safety` hook also requires `poetry` to be in your `PATH`.
> > Ensure the poetry installed with the same python version as in `pyproject.toml`

Reviewer: Maybe we should add poetry installation link here? 🤔 
```suggestion
> The `safety` hook also [requires](https://github.com/Lucas-C/pre-commit-hooks-safety?tab=readme-ov-file#supported-files) `poetry` to be in your `PATH`.
> Ensure the poetry [installed](https://python-poetry.org/docs/#installation) with the same python version as in `pyproject.toml`
```
